core/services/relay/evm/mercury/v4/reportcodec/report_codec_test.go

```diff
@@ -0,0 +1,163 @@
+		assert.Equal(t, reportElems["bid"].(*big.Int).Int64(), int64(244))
+		assert.Equal(t, reportElems["ask"].(*big.Int).Int64(), int64(245))
+		assert.Equal(t, reportElems["validFromTimestamp"].(uint32), uint32(123))
+		assert.Equal(t, reportElems["expiresAt"].(uint32), uint32(20))
+		assert.Equal(t, reportElems["linkFee"].(*big.Int).Int64(), int64(456))
+		assert.Equal(t, reportElems["nativeFee"].(*big.Int).Int64(), int64(457))
+		assert.Equal(t, reportElems["marketStatus"].(uint32), uint32(1))
+
+		assert.Equal(t, types.Report{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7b, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc9, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x14, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1}, report)
+		max, err := r.MaxReportLength(4)
+		require.NoError(t, err)
+		assert.LessOrEqual(t, len(report), max)
+
+		// t.Run("Decode decodes the report", func(t *testing.T) {
+		// 	decoded, err := r.Decode(report)
+		// 	require.NoError(t, err)
+		//
+		// 	require.NotNil(t, decoded)
+		//
+		// 	assert.Equal(t, uint32(242), decoded.ObservationsTimestamp)
+		// 	assert.Equal(t, big.NewInt(243), decoded.BenchmarkPrice)
+		// 	assert.Equal(t, big.NewInt(244), decoded.Bid)
+		// 	assert.Equal(t, big.NewInt(245), decoded.Ask)
+		// 	assert.Equal(t, uint32(123), decoded.ValidFromTimestamp)
+		// 	assert.Equal(t, uint32(20), decoded.ExpiresAt)
+		// 	assert.Equal(t, big.NewInt(456), decoded.LinkFee)
+		// 	assert.Equal(t, big.NewInt(457), decoded.NativeFee)
+		// 	assert.Equal(t, uint32(1), decoded.MarketStatus)
+		// })
```

Comment:
commented code?

Reply:
My mistake, did not mean to comment this out.